### DISCUSSION TITLE
touch control

### DISCUSSION THREAD

**Discussion by CATdj** [Kitware/trame#362] [General]
Hi, I setup a model using PyVistaRemoteLocalView but cannot zoom in/out with 2 finger pinch on touch screen. I tried the example with VtkLocalView, it works. What could be done to make PyVistaRemoteLocalView work as well? Single finger rotation works though. Thanks.

**Comment by jourdain** [comment]
Yes, when remote rendering mouse/touch mapping is missing some proper handling, but we are [working on it](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/10545) via [trame-rca](https://github.com/Kitware/trame-rca/).

**Reply by CATdj** [reply]
Thank you! do you have a rough timeline when this will be implemented?

**Reply by jourdain** [reply]
The implementation is there and will be merged pretty soon. 

But then you will need trame-vtk to update to start leveraging it.
Then some PyVista edits if you want to make it dynamic/available sooner. 

So it will take some time to be mainstream. But if you have funding, we can probably push all of that faster or at least come up with a path on your side so you can leverage it within a week or so.

**Reply by CATdj** [reply]
thanks. How much would it be?

**Reply by jourdain** [reply]
It will depend on your stack and how much involvement it will be to enable it within your app. You should reach out to [Kitware](https://www.kitware.com/contact/) so we can assess.
